services/wallet/router/pathprocessor/processor_bridge_hop.go

```diff
@@ -201,7 +201,7 @@ func (h *HopBridgeProcessor) packTxInputDataInternally(params ProcessorInputPara
 		return []byte{}, createBridgeHopErrorResponse(err)
 	}
 
-	bonderKey := makeKey(params.FromChain.ChainID, params.ToChain.ChainID, "", "")
+	bonderKey := makeKey(params.FromChain.ChainID, params.ToChain.ChainID, "", "", params.AmountIn)
```

Comment:
not sure what you want to do here.
You want paths from a prior route calculation not to be used when the user changes the amount in the Send/Bridge/Swap popup? If that's the case:
- Shouldn't these tmp values be reset with some other external logic?
- Why not add uuid to the key to identify paths instead? We should be using a different one on each request.

Reply:
> You want paths from a prior route calculation not to be used when the user changes the amount in the Send/Bridge/Swap popup? 

They are certainly not used cause we do call `Clear()` for each pathprocessor.

> Why not add uuid to the key to identify paths instead? We should be using a different one on each request to ensure we don't mix results.

Results are not mixed cause we clear the state of every pathprocessor on each request.

> Shouldn't these tmp values be reset with some other external logic whenever the user changes the input value?

No, it doesn't need to be reset. Depending on the router input parameters we may have a lot of bridge paths in the candidates list, we need to identify each of them uniquely and that's why the amount is added to the key, but previously that was done using just "from" and to "chain" and that was not enough, cause if the user has positive balances across different chains we're preparing different options for taking certain amount from different chains and the rest from other chain/s and then searching for the best route among them. In the end the best route may have one or more bridge paths, but when sending using just from and to chain doesn't take appropriate bonder data, that's why we've added the amount to the list (key composed of fromChain-toChain-amount is now unique).